test/test_common/utility.cc

```diff
@@ -207,6 +207,17 @@ TestHeaderMapImpl::TestHeaderMapImpl(
 
 TestHeaderMapImpl::TestHeaderMapImpl(const HeaderMap& rhs) : HeaderMapImpl(rhs) {}
 
+TestHeaderMapImpl& TestHeaderMapImpl::operator=(const TestHeaderMapImpl& other) {
+  if (&other == this) {
+    return *this;
+  }
+
+  removePrefix(LowerCaseString(""));
```

Comment:
I would rather not make any API public on `HeaderMapImpl` that we don't need to. This is a performance sensitive object and we should avoid having folks do slow/inefficient things with it that they don't need to.

Reply:
If someone does need to clear(), I think it'd be better for it to be abstracted at the class interface. 
Right now removePrefix(LowerCaseString("")) seems like it might be specific to the implementation?

But a compromise might be to define clear() but make it private or protected, with a note about exposing it in the future if there's a real need.

Reply:
Arguably, nobody should be doing this, `removePrefix` is a horrible way to clear, it's only suitable for test purposes. I'm going to follow Matt's suggestion and move this to `TestHeaderMapImpl` only.